lib/ts/recipe/session/index.ts

```diff
@@ -304,24 +332,36 @@ export default class SessionWrapper {
         });
     }
 
-    static refreshSession(req: any, res: any, userContext: any = {}) {
+    static async refreshSession(req: any, res: any, userContext: any = {}) {
         const recipeInstance = Recipe.getInstanceOrThrowError();
         const config = recipeInstance.config;
         const recipeInterfaceImpl = recipeInstance.recipeInterfaceImpl;
 
-        return refreshSessionInRequest({ res, req, userContext, config, recipeInterfaceImpl });
+        return await refreshSessionInRequest({ res, req, userContext, config, recipeInterfaceImpl });
     }
 
-    static refreshSessionWithoutRequestResponse(
+    static async refreshSessionWithoutRequestResponse(
         refreshToken: string,
         disableAntiCsrf: boolean = false,
         antiCsrfToken?: string,
+        antiCSRF: "VIA_TOKEN" | "VIA_CUSTOM_HEADER" | "NONE" | undefined = undefined,
         userContext: any = {}
     ) {
+        const recipeInstance = Recipe.getInstanceOrThrowError();
+        const appInfo = recipeInstance.getAppInfo();
+
+        if (antiCSRF === undefined) {
```

Comment:
Maybe this should happen inside `refreshSession`? Im not a fan of an index function throwing validation errors like this. Moving it into the function would also remove the need for this function to be marked as async